sgf/parser.go

```diff
@@ -0,0 +1,174 @@
+package sgf
+
+import (
+	"errors"
+	"io"
+)
+
+// A node is comprised of key-value pairs (might hold other info someday)
+type Node struct {
+	Fields map[string]string
+}
+
+func NewNode() *Node {
+	n := &Node{}
+	n.Fields = make(map[string]string)
+	return n
+}
+
+// A game record (sgf) struct for now is just a collection of nodes
+// TODO: this struct should retain the branch structure of game variations
+type Sgf struct {
+	Index int
+	Nodes map[int]*Node
```

Comment:
Typically, I would expect a Game to have a single Root node.

Reply:
Agreed, fixed